SpatialGDK/Source/SpatialGDKServices/Private/LocalReceptionistProxyServerManager.cpp

```diff
@@ -0,0 +1,265 @@
+	return true;
+}
+
+void FLocalReceptionistProxyServerManager::Init(int32 Port)
+{
+	if (!IsRunningCommandlet())
+	{
+		LocalReceptionistProxyServerPreRunChecks(Port);
+	}
+}
+
+
+bool FLocalReceptionistProxyServerManager::TryStopReceptionistProxyServer()
+{
+	if (ProxyServerProcHandle.IsValid())
+	{
+		SpatialCommandUtils::StopLocalReceptionistProxyServer(ProxyServerProcHandle);
+		bProxyIsRunning = false;
+		return true;
+	}
+
+	return false;
+}
+
+
+TSharedPtr<FJsonObject> FLocalReceptionistProxyServerManager::ParsePidFile()
+{
+	FString ProxyInfoFileResult;
+	TSharedPtr<FJsonObject> JsonParsedProxyInfoFile;
+
+	if (FFileHelper::LoadFileToString(ProxyInfoFileResult, *ProxyInfoFilePath))
+	{
+		if (FSpatialGDKServicesModule::ParseJson(ProxyInfoFileResult, JsonParsedProxyInfoFile))
+		{
+			return JsonParsedProxyInfoFile;
+		}
+		else
+		{
+			UE_LOG(LogLocalReceptionistProxyServerManager, Error, TEXT("Json parsing of proxyInfo.json failed. Can't get proxy's Pid."));
+		}
+	}
+	else
+	{
+		UE_LOG(LogLocalReceptionistProxyServerManager, Error, TEXT("Loading proxyInfo.json failed. Can't get proxy's Pid."));
+	}
+
+	return nullptr;
+}
+
+void FLocalReceptionistProxyServerManager::SetPidInJson(const FString& Pid)
+{
+	FString ProxyInfoFileResult;
+
+	// If file does not exist create an empty json file 
+	if (!FPaths::FileExists(ProxyInfoFilePath))
+	{
+		FFileHelper::SaveStringToFile(TEXT("{ }"), *ProxyInfoFilePath);
+	}
+
+	TSharedPtr<FJsonObject> JsonParsedProxyInfoFile = ParsePidFile();
+	if (!JsonParsedProxyInfoFile.IsValid())
+	{
+		UE_LOG(LogLocalReceptionistProxyServerManager, Error, TEXT("Failed to update Pid(%s). Please ensure that the following file exists: %s"), *Pid, *SpatialGDKServicesConstants::ProxyInfoFilename);
+		return;
+	}
+
+	JsonParsedProxyInfoFile->SetStringField("pid", Pid);
+
+	TSharedRef<TJsonWriter<>> JsonWriter = TJsonWriterFactory<>::Create(&ProxyInfoFileResult);
+	if (!FJsonSerializer::Serialize(JsonParsedProxyInfoFile.ToSharedRef(), JsonWriter))
+	{
+		UE_LOG(LogLocalReceptionistProxyServerManager, Error, TEXT("Failed to write pid to parsed proxy info file. Unable to serialize content to json file."));
+		return;
+	}
+	if (!FFileHelper::SaveStringToFile(ProxyInfoFileResult, *FPaths::Combine(SpatialGDKServicesConstants::ProxyFileDirectory, SpatialGDKServicesConstants::ProxyInfoFilename)))
+	{
+		UE_LOG(LogLocalReceptionistProxyServerManager, Error, TEXT("Failed to write file content to %s"), *SpatialGDKServicesConstants::ProxyInfoFilename);
+	}
+}
+
+FString FLocalReceptionistProxyServerManager::ParsePid()
+{
+	FString Pid;
+
+	if (TSharedPtr<FJsonObject> JsonParsedProxyInfoFile = ParsePidFile())
+	{
+		if (JsonParsedProxyInfoFile->TryGetStringField(TEXT("pid"), Pid))
+		{
+			return Pid;
+		}
+		else
+		{
+			UE_LOG(LogLocalReceptionistProxyServerManager, Error, TEXT("'pid' does not exist in proxyInfo.json. Can't read proxy's Pid."));
+		}
+	}
+
+	Pid.Empty();
```

Comment:
I think I would prefer to change the function to be `TryParsePid` and then return true/false and the pid as an out argument.
just to make it very clear that this function can fail to parse the pid